File: android/app/src/main/java/com/albbiz/map/data/ClaimRequest.kt
// Bismillah Hir Rahman Nir Raheem
package com.albbiz.map.data

data class ClaimRequest(
    val id: String = "",
    val businessId: String = "",
    val businessName: String = "",
    val userId: String = "",
    val userName: String = "",
    val userEmail: String = "",
    val reason: String = "",
    val status: String = "pending", // pending, approved, rejected
    // "claim": user doesn't yet own this listing, wants ownerId reassigned to them.
    // "verification": user already owns this listing, just wants isVerified set —
    // approving still calls the same ownerId-reassign code, but it's a no-op since
    // userId already equals the business's current ownerId.
    val type: String = "claim",
    val createdAt: Long = System.currentTimeMillis()
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "id" to id,
            "businessId" to businessId,
            "businessName" to businessName,
            "userId" to userId,
            "userName" to userName,
            "userEmail" to userEmail,
            "reason" to reason,
            "status" to status,
            "type" to type,
            "createdAt" to createdAt
        )
    }
}

File: ios/AlbBizMap_IOS/AlbBizMap/Models/ClaimRequest.swift
// Bismillah Hir Rahman Nir Raheem
import Foundation

public struct ClaimRequest: Identifiable, Codable, Hashable {
    public var id: String
    public var businessId: String
    public var businessName: String
    public var userId: String
    public var userName: String
    public var userEmail: String
    public var reason: String
    public var status: String // pending, approved, rejected
    public var createdAt: Int64
    
    public init(
        id: String = "",
        businessId: String = "",
        businessName: String = "",
        userId: String = "",
        userName: String = "",
        userEmail: String = "",
        reason: String = "",
        status: String = "pending",
        createdAt: Int64 = Int64(Date().timeIntervalSince1970 * 1000)
    ) {
        self.id = id
        self.businessId = businessId
        self.businessName = businessName
        self.userId = userId
        self.userName = userName
        self.userEmail = userEmail
        self.reason = reason
        self.status = status
        self.createdAt = createdAt
    }
    
    public func toMap() -> [String: Any?] {
        return [
            "id": id,
            "businessId": businessId,
            "businessName": businessName,
            "userId": userId,
            "userName": userName,
            "userEmail": userEmail,
            "reason": reason,
            "status": status,
            "createdAt": createdAt
        ]
    }
    
    public static func fromMap(id: String, map: [String: Any?]) -> ClaimRequest {
        return ClaimRequest(
            id: id,
            businessId: map["businessId"] as? String ?? "",
            businessName: map["businessName"] as? String ?? "",
            userId: map["userId"] as? String ?? "",
            userName: map["userName"] as? String ?? "",
            userEmail: map["userEmail"] as? String ?? "",
            reason: map["reason"] as? String ?? "",
            status: map["status"] as? String ?? "pending",
            createdAt: (map["createdAt"] as? NSNumber)?.int64Value ?? Int64(Date().timeIntervalSince1970 * 1000)
        )
    }
}
